### DISCUSSION TITLE
ColorCorrectionModel

### DISCUSSION THREAD

**Discussion by JeeDevUser** [bytedeco/javacpp#568] [General]
Hi,

I am just trying to implement this sample code to the Java:

https://docs.opencv.org/4.x/d1/dc1/tutorial_ccm_color_correction_model.html

but corresponding Java class for **_ColorCorrectionModel_**
https://docs.opencv.org/4.x/df/d38/classcv_1_1ccm_1_1ColorCorrectionModel.html

 seems to be missing. Whether I missed something, or this example cannot be transferred to Java?

**Comment by saudet** [comment]
It just looks like the `opencv2/mcc/ccm.hpp` header file was missing from the presets.
I've fixed that in commit https://github.com/bytedeco/javacpp-presets/commit/f575bf0739f425b7e8d210555a77e9b2b3b17f86.
Please give it a try with the snapshots: http://bytedeco.org/builds/
Thank you for reporting this issue!

**Reply by JeeDevUser** [reply]
Thanks for response!
I'm still new to this, and I need your help, I'm a little confused about which jar file to download :-(,  can you help?
as far as I can see, the files are somewhere here (correct me if I'm wrong), I am not using maven 

https://oss.sonatype.org/content/repositories/snapshots/org/bytedeco/opencv/4.5.5-1.5.8-SNAPSHOT/

Now, I'm making an OpenCV application that should work on Win, Linux and Android.
can you specify exactly which files I need to download for each of these platforms?

Thanks in advance!

**Reply by saudet** [reply]
Well, if you really want to do it manually, you'll need to get all the latest JAR files (the ones with the newest biggest version numbers) from these directories:
https://oss.sonatype.org/content/repositories/snapshots/org/bytedeco/opencv/4.5.5-1.5.8-SNAPSHOT/
https://oss.sonatype.org/content/repositories/snapshots/org/bytedeco/openblas/0.3.20-1.5.8-SNAPSHOT/
https://oss.sonatype.org/content/repositories/snapshots/org/bytedeco/javacpp/1.5.8-SNAPSHOT/

**Reply by JeeDevUser** [reply]
OMG, there are a lot of them ;-)
But in the official release, there will be option to download it manually, as usual, from 

https://github.com/bytedeco/javacpp
https://github.com/bytedeco/javacv/

is not? 
Do you have any idea when the official release will be out?

P.S. otherwise, what you are doing is truly extraordinary, and deserves every possible praise, I honestly think so !

**Reply by saudet** [reply]
These new wrappers have been released with version 1.5.8! Thanks for reporting

**Reply by JeeDevUser** [reply]
Hi, thanks for the tip!
> These new wrappers have been released with version 1.5.8! Thanks for reporting


with 1.5.8 or 1.5.9?
And what is maven content now?
For snapshot , this was
http://bytedeco.org/builds/

**Reply by saudet** [reply]
What do you want to know exactly? They are not going to get removed from 1.5.9, if that's what you're asking.
